pages/blog/typescript-types.mdx

```diff
@@ -0,0 +1,159 @@
+import withArticle from '../../ui/blog/article'
+export const meta = {
+  title: 'TypeScript basic types system',
+  author: 'tuvalSimha',
+  tags: ['graphql', 'typescript', 'codegen'],
+  date: '2022-07-16',
+  description: 'How to use Typescript basic types system',
+  image: '/blog-assets/graphql-code-generator/cover.png'
+}
+export default withArticle(meta)
+ 
+ 
+# Introduction
+
+TypeScript programming language is a superset of JavaScript that adds types to JavaScript using a set of tools called a type system.
+All types inference assumes the expected type of the variable on code time. Type inference will log a complaint if the variable is later reassigned to a different type.
+For Example:
+
+```ts
+let pizzaName = 'Napolitana'
+
+pizzaName = 123 // Error: Type number (123) is not assignable to type 'string'.
+```
+
+When a variable is declared without being assigned an initial value, the default type is `any`.
+
+```ts
+let pizzaName
+
+pizzaName = 123 // Any got all types of values
+pizzaName = 'Napolitana' // Any got all types of values
+```
+
+Adding a type annpotation to a variable will force the variable to be of that type. It help you make sure the output will be the expected. And if not, you will get types error in code-time.
+
+```ts
+let pizzaName: string
+pizzaName = 'Napolitana'
+
+pizzaName = 123 // Error: Type number (123) is not assignable to type 'string'.
+```
+
+## Union Types
+
+After we understand how TypeScript type system works, we can start to understand how to use it to create a Union Type.
+TypeScript allow us a flexible type that called `any` type like we did in the previous example. But also TypeScript can combine multiple types and combine them together as a union type.
+
+```ts
+let pizzaLover: string | boolean
+
+pizzaLover = 'YES' // string type
+pizzaLover = true // boolean type
+```
+
+More Union type example:
```

Comment:
I thikn the jump from `string | boolean` to `A | B` is fine, but you should also mention here that TS supports string unions as well and give an example. 

```ts
type PizzaAddons = "olives" | "mushrooms" | "pepperoni";

const myPreference: PizzaAddons = "olives"; // ok
const myPreference: PizzaAddons = "pineapple"; // error
```